impl/src/test/java/io/kroxylicious/testing/kafka/common/KafkaClusterTest.java

```diff
@@ -67,30 +69,44 @@ public void kafkaClusterZookeeperMode() throws Exception {
         }
     }
 
-    @Test
-    public void kafkaTwoNodeClusterKraftMode() throws Exception {
-        int brokersNum = 2;
+    @TestTemplate
+    @ExtendWith(KafkaClusterTestInvocationContextProvider.class)
+    public void kafkaClusterTemplate(KafkaClusterTestCase testCase) throws Exception {
```

Comment:
The way the dependencies worked (in my head at least) was that the `KafkaClusterFactory`, `KafkaClusterConfig`  `KafkaClusterConfig` and related classes are lower level, and don't have a dependency on the annotations. This test was respecting that layering, but adding the extension mixes things up. Can we keep this test as it was, and add a new test, just for the `@Version` annotation?

Reply:
done